src/api/crypto/blockchain/BalanceOracle.cpp

```diff
@@ -311,30 +292,24 @@ auto BalanceOracle::Imp::process_update_balance(
     Balance input) noexcept -> void
 {
     auto changed{false};
-    auto& data = [&]() -> auto&
-    {
-        if (auto i = data_.find(chain); i != data_.end()) {
-            auto& nymData = i->second.second;
-
-            if (auto j = nymData.find(owner); j != nymData.end()) {
-                auto& out = j->second;
-                changed = (out.first != input);
+    Data* data;
```

Comment:
As a minimum, initialize data, or use a long and easy to debug option as suggested in the earlier overload of process_update_balance.